platform/mbed_retarget.cpp

```diff
@@ -826,8 +826,13 @@ void mbed_set_unbuffered_stream(std::FILE *_file) {
  */
 std::FILE *mbed_fdopen(FileHandle *fh, const char *mode)
 {
-    char buf[12]; /* :0x12345678 + null byte */
-    std::sprintf(buf, ":%p", fh);
+    // This is to avoid scanf(buf, ":%.4s", fh) and the bloat it brings.
+    char buf[2 + sizeof(fh) + 1]; /* :(pointer) + null byte */
```

Comment:
The NULL terminator is not required. `buf` is not a NULL terminated string it is rather a pointer address prefixed with the character `:`.
I'm also wondering why the buffer have to be 7 byte long. 5 byte should be sufficient.